pkg/operator/staticpod/startupmonitor/fallback.go

```diff
@@ -128,7 +136,47 @@ func (f *staticPodFallback) fallbackToPreviousRevision(reason, message string) e
 }
 
 func (f *staticPodFallback) markRevisionGood() error {
-	return f.createLastKnowGoodRevisionFor(f.revision, true)
+	if err := f.createLastKnowGoodRevisionFor(f.revision, true); err != nil {
+		return err
+	}
+
+	if f.operatorClient == nil || len(f.nodeName) == 0 {
+		return nil
+	}
+
+	// the startup-monitor is authorative to signal readiness of a revision. The operator
+	// is waiting for this to happen. Otherwise, the operator could assume readiness while
+	// the startup-monitor falls back, leading to an awkward situation.
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
what will the installer do when we fail to update the status?
If I am reading the code right, it will indefinitely wait.

Reply:
in the worst case the operator is Progressing forever, right?

Not sure how we can get rid of this race. We can make the retry here loop forever (before suicice).

Reply:
done